docs/source/rule-testing.rst

```diff
@@ -1,30 +1,100 @@
-
-  {
-    "records": [
-      {
-        "data": {} or "",
-        "description": "information about this test",
-        "log": "log_type_from_logs.json",
-        "service": "kinesis" or "s3" or "sns" or "stream_alert_app",
-        "source": "kinesis_stream_name" or "s3_bucket_id" or "sns_topic_name" or "stream_alert_app_function_name",
-        "trigger_rules": [
-          "rule_01",
-          "rule_02"
-        ]
-      }
-    ]
-  }
-
-.. note:: Multiple tests can be included in one file simply by adding them to the "records" array within the `name_of_your_tests.json` file.
+To test a new rule, first create a new JSON file anywhere within the 'tests/integration/rules' directory with the ``.json`` extension.
+
+This file should contain the following structure:
+
+.. code-block:: json
+
+  [
+    {
+      "data": "Either a string, or JSON map",
+      "description": "This test should trigger or not trigger an alert",
+      "log": "The log name declared in logs.json",
+      "service": "The service sending the log - kinesis, s3, sns, or stream_alert_app",
+      "source": "The exact resource which sent the log - kinesis stream name, s3 bucket ID, SNS topic name, or stream_alert_app_function name",
+      "trigger_rules": [
+        "list of rule names which should generate alerts",
+        "another potential rule name could go here"
+      ]
+    }
+  ]
+
+.. note:: Multiple tests can be included in one file by adding them to the array above.
+
+When specifying the test data, it can be either of two fields:
+
+1. ``"data"``: An entire example record, with all necessary fields to properly classify
+2. ``"override_record"``: A subset of the example record, where only relevant fields are populated
+
+The advantage of option #2 is that the overall test event is much smaller.
+
+The testing framework will auto-populate the records behind the scenes with the remainder fields for that given log type.
+
+For example:
+
+.. code-block:: json
+
+  [
+    {
+      "data": {
+        "account": "123456789102",
+        "detail": {
+          "request": {
+            "eventName": "putObject",
+            "bucketName": "testBucket"
+          }
+        },
+        "detail-type": "API Call",
+        "id": "123456",
+        "region": "us-west-2",
+        "resources": [
+          "testBucket"
+        ],
+        "source": "aws.s3",
+        "time": "Jan 01 2018 12:00",
+        "version": "1.05"
+      },
+      "description": "An example test with a full cloudwatch event",
+      "log": "cloudwatch:events",
+      "service": "s3",
+      "source": "test-s3-bucket-name",
+      "trigger_rules": [
+        "my_fake_rule"
+      ]
+    }
+  ]
+
+Let's say a rule is only checking the value of ``source`` in the test event.  In that case, it's not valuable to fill in all the other data.  Here is what the event would  look like with ``override_record``:
```

Comment:
there is an extra space between  `would` and `look`.

I'd suggest changing `it's not valuable to fill in all the other data` to something like `there is no added benefit to filling in all of the other data`